File: app/src/test/java/com/tomaszrykala/dorefindmi/things/supplier/ledstrip/MockLedStripSupplier.java
package com.tomaszrykala.dorefindmi.things.supplier.ledstrip;

import com.tomaszrykala.dorefindmi.model.Led;

public class MockLedStripSupplier implements LedStripSupplier {

    @Override public void init() {
        System.out.println("MockLedStripSupplier.init");
    }

    @Override public boolean isLitAt(Led led) {
        System.out.println("MockLedStripSupplier.isLitAt");
        return true;
    }

    @Override public void light(Led led) {
        System.out.println("MockLedStripSupplier.light");
    }

    @Override public void reset() {
        System.out.println("MockLedStripSupplier.reset");
    }

    @Override public void close() throws Exception {
        System.out.println("MockLedStripSupplier.close");
    }
}

File: app/src/test/java/com/tomaszrykala/dorefindmi/things/supplier/ledstrip/MockLedStripSupplier.kt
package com.tomaszrykala.dorefindmi.things.supplier.ledstrip

class MockLedStripSupplier : LedStripSupplier {

    override fun light(colors: IntArray) {
        println("MockLedStripSupplier.light")
    }

    override fun getLength(): Int {
        println("MockLedStripSupplier.getLength")
        return 7
    }

    @Throws(Exception::class)
    override fun close() {
        println("MockLedStripSupplier.close")
    }
}
